### PR TITLE
Add instructions for settings up Android Jenkins projects

### PR DESCRIPTION
**Pull request by f2rf2r**
Should this live here or in https://github.com/IntrepidPursuits/shhherpa?

### CODE REVIEW

File: android/android_jenkins.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Setting up Jenkins CI for Android
+
+### Templates
+There are three templates for creating Android Jenkins projects:
+* `android-template`: This is runs every time something is merged into the develop branch. It runs the unit tests, generates a QA build, and uploads it to the OTA link.
```
Comment (mpeng3) [inline android/android_jenkins.md]: Typo - should be "This is run"

File: android/android_jenkins.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* `android-pr-template`: This is run every time a PR is created. It runs the unit tests and reports the result back to the GitHub PR page.
+* `android-nightly-template`: This is run once every weeknight between 12am and 8am. It runs both the unit tests and UI tests (currently the UI tests will run on a physical device attached to the Jenkins machine) and reports the test coverage.
+
+### Setup instructions for all templates
+1. In the Jenkins homepage, click `New Item`
+1. Enter the project name. The name should be in the form of `project-android`, `project-android-pr`, or `project-android-nightly`.
+1. In the `Copy from` box, enter one of the template names above.
+1. Click OK, and you will be directed to the project configuration page.
+1. Under the `Source Code Management` settings, set the proper `Repository URL` (this is basically the same as the Project url, but might be in slightly different format)
+1. Change the `Branches to build` if your manin branch is not master.
```
Comment (mpeng3) [inline android/android_jenkins.md]: Typo - should be "main branch"

File: android/android_jenkins.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+1. Choose a branch for `Protected branches` (usually master or develop)
+1. Click the `Edit` button for that branch.
+1. Check `Protect this branch` and `Require status checks to pass before merging`
+1. Click `Save changes`
+1. You should be all set! Jenkins will now build every PR (including every time the PR is updated), and will show the build status on the PR page.
+
+Note: if you wish to rebuild a PR without pushing new commits, you can type `restart jenkins build` as a comment on the Github PR page to force a rebuild.
+
+### Additional instructions for `android-nightly-template`
+The `android-night-template` expects the project to contain [coverage.gradle](https://github.com/IntrepidPursuits/AndroidSkeleton/blob/master/app/coverage.gradle). If your project is based off [Android Skeleton](https://github.com/IntrepidPursuits/AndroidSkeleton), you should be all set. If not, you can simply copy that file to your project and add `apply from: 'coverage.gradle'` in your app `build.gradle` file (note that you will also need the [spoon plugin](https://github.com/stanfy/spoon-gradle-plugin)).
```
Comment (mpeng3) [inline android/android_jenkins.md]: Typo - template name should be "android-nightly-template", right?

**Review by mpeng3 — APPROVED** [review]
A few typos, but otherwise, :shipit:. Thank you for posting this info! 👍

File: android/android_jenkins.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (prfarlow1) [inline android/android_jenkins.md]: What's going on here

File: android/android_jenkins.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+ss
+s
+s
```
Comment (prfarlow1) [inline android/android_jenkins.md]: 🐍

**Review by prfarlow1 — CHANGES_REQUESTED** [review]
Lots of trailing whitespace?

File: android/android_jenkins.md (hunk 4 above)
Comment (f2rf2r) [reply android/android_jenkins.md]: 🐍

**Review by prfarlow1 — APPROVED** [review]
:shipit:

**Comment by deanshi** [comment]
:shipit:

**Pull request merged**